chart/flux/values.yaml

```diff
@@ -275,6 +277,14 @@ kube:
 # additionalArgs:
 # - --connect=ws://fluxcloud
 
+# The contents of the secret will be defined as environment variables
+# in the Flux container (supported key values by Flux for now are
+# `GIT_AUTHUSER` and `GIT_AUTHKEY`). Once defined, you can use the
+# variables in your `git.url`:
+# `https://$(GIT_AUTHUSER):$(GIT_AUTHKEY)@github.com/fluxcd/flux-get-started.git`
```

Comment:
What happens if someone only wants to supply the authkey, as shown in https://github.com/fluxcd/flux/pull/2438#issuecomment-530265107?

Reply:
This would never work in any setup. Although git supports only passing along the username (and fetching the actual token from some keystore), passing along a _token_ as first parameter before the `@` will always result in it being parsed as an username.

In fact, passing along any credentials in an URL has been [deprecated with RFC3986](https://tools.ietf.org/html/rfc3986#section-3.2.1) but is backported / still supported by many.

Reply:
> passing along a _token_ as first parameter before the `@` will always result in it being parsed as an username

I would have expected this as well. Can you explain the example given in that comment? Is it invalid?

Reply:
I think it comes from not following the GitHub documentation closely, or not being familiar with basic auth.

We could theoretically block things if we detect that only a `GIT_AUTHKEY` is present in the given `--git-url`; but I am not sure if that is worth the technical debt it would introduce.

Reply:
I couldn't find an example _in the github docs_ that showed a key being used in username position. So I think it's a safe course of action to assume a username will be provided.